src/Lykke.Job.CandleHistoryWriter.Repositories/HistoryMigration/HistoryProviders/TradesSQLHistory/TradesSqlHistoryRepository.cs

```diff
@@ -100,10 +153,12 @@ private string BuildCurrentQueryCommand()
             var commandBld = new StringBuilder();
 
             // TODO: implement a query with pagination.
-            commandBld.Append($@"SELECT Id, Volume, Price, ""DateTime"", OppositeVolume ");
+            commandBld.Append($@"SELECT Id, (Asset + OppositeAsset) AS AssetToken, Direction, Volume, Price, ""DateTime"", OppositeVolume, OrderId, OppositeOrderId, TradeId ");
             commandBld.Append("FROM Trades ");
             commandBld.Append("WHERE ");
-            commandBld.Append($@"OrderType = 'Limit' AND Asset + OppositeAsset = '{AssetPairId}' AND Direction IN ('Buy', 'Sell') ");
+            commandBld.Append($@"OrderType = 'Limit' AND (Asset + OppositeAsset = '{SearchToken}' OR OppositeAsset + Asset = '{SearchToken}') AND Direction IN ('Buy', 'Sell') ");
```

Comment:
Use SQL arguments. They work better if you execute the same query multiply times.
I would added columns for Asset + OppositeAsset and OppositeAsset + Asset and then created indices for them. It would dramatically improved performance.

Reply:
Better indices are OrderType  + (Asset + OppositeAsset ) and (OppositeAsset + Asset) + Direction

Reply:
It's not so easy to perform it 'cause I'm not that person who owns the DB. But thank you, I'll try to  arrange these indices creation if we will run into a disastrously low performance of sql query.

Reply:
There was also comment about using SQL arguments rather then dynamic SQL command. https://stackoverflow.com/a/1608856/8592182